src/librbd/image/CloneRequest.cc

```diff
@@ -124,6 +124,13 @@ void CloneRequest<I>::validate_options() {
     }
```

Comment:
What about
```
  if (format < 2) {
    lderr(m_cct) << "format 2 or later required for clone" << dendl;
    complete(-EINVAL);
    return;
  }
```
check above?  I think it should also become
```
  if (format < 2) {
    lderr(m_cct) << "unsupported format for clone: " << format << dendl;
    complete(-EINVAL);
    return;
  }
```
to avoid accepting `RBD_IMAGE_OPTION_FORMAT` with to a bogus value on cloning.

Reply:
LGTM, It is also consistent with other checks

Reply:
Sorry, it seems I goofed up the comment.  I meant
```
  if (format != 2) {
    lderr(m_cct) << "unsupported format for clone: " << format << dendl;
    complete(-EINVAL);
    return;
  }
```
of course -- the point is not accept 0 or 3 here.

Reply:
ok